src/assets/CollectiblesController.test.ts

```diff
@@ -102,7 +107,29 @@ describe('CollectiblesController', () => {
     nock.enableNetConnect();
   });
 
+  let harness: Harness;
+  let collectiblesController: CollectiblesController;
+  let mockCollectible: Collectible;
+  let selectedAddress: string;
+  let chainId: string;
+  let assetsContract: AssetsContractController;
+
   beforeEach(() => {
```

Comment:
This PR seems to extend the use of `beforeEach` so that the "setup" portion of each test does not have to be explicitly specified. While I realize that this approach may seem beneficial, because it seems DRY, I personally think it does more harm than good in the long run. There are a number of people who have written about overapplying the DRY principle in tests — [here][1] are a [couple][2] that I like — but the main point is that this style of writing tests makes it difficult to look at a single test and understand how it works in isolation from setup to assertion. It's typical when using hooks to assign a bunch of variables like we are doing here, but that makes it difficult to understand which setup data is important to a given test and which is not. It also means that if another `beforeEach` were to be specified in a nested `describe` block down the road, then it would be difficult to piece together the full set of setup data, because one would have to mentally combine those variables. Finally, it couples tests together in a way that makes it difficult to customize a test later. Each test will have its own needs, and while some needs are shared, others are not, so being able to understand the key differences between each test, and specify the exact needs of any one test, is important. 

Instead of hooks, I generally advise the use of helper functions. I see that such a function, `setupController`, already exists in this file, but I think that is perfectly fine to call this function at the start of each test. Let's be honest, these tests could use work, and there are a number of things we can do to make them more understandable and maintainable, but I'm not so sure that this would move things in a more desirable direction. 

That's my two cents, but curious to know what others think.

Reply:
Agreed! I find tests that use setup functions to be far easier to read than `beforeEach, and I have encountered these `beforeEach`-related maintenance headaches too many times to count.